src/main/java/org/embulk/input/mongodb/MongodbInputPlugin.java

```diff
@@ -324,10 +330,52 @@ private MongoClient createClientFromParams(PluginTask task)
         }
 
         if (task.getUser().isPresent()) {
-            return new MongoClient(addresses, Arrays.asList(createCredential(task)));
+            return new MongoClient(addresses, Arrays.asList(createCredential(task)), createMongoClientOptions(task));
         }
         else {
-            return new MongoClient(addresses);
+            return new MongoClient(addresses, createMongoClientOptions(task));
+        }
+    }
+
+    private MongoClientOptions createMongoClientOptions(PluginTask task)
+    {
+        MongoClientOptions.Builder builder = new MongoClientOptions.Builder();
+        if (task.getTls()) {
+            builder.sslEnabled(true);
+            if (task.getTlsInsecure()) {
+                builder.sslInvalidHostNameAllowed(true);
+                builder.sslContext(createAcceptAllCertificate());
+            }
+        }
+        return builder.build();
+    }
+
+    private SSLContext createAcceptAllCertificate()
```

Comment:
nitpick: The method name is difficult to understand: verb (create) + verb (accept).

How about: `createSSLContextToAcceptAnyCert` ?